packages/datatrak-web/src/api/mutations/useEditUser.ts

```diff
@@ -8,20 +8,25 @@ import { Project } from '@tupaia/types';
 import { put } from '../api';
 import { Entity } from '../../types';
 
-type UserPreferences = { projectId?: Project['id']; countryId?: Entity['id'] };
+type UserPreferences = {
+  projectId?: Project['id'];
+  countryId?: Entity['id'];
+  deleteAccountRequested?: boolean;
+};
 
-export const useEditUser = onSuccess => {
+export const useEditUser = (onSuccess?: () => void) => {
   const queryClient = useQueryClient();
 
   return useMutation<any, Error, UserPreferences, unknown>(
-    async ({ projectId, countryId }: UserPreferences) => {
-      if (!projectId && !countryId) {
+    async ({ projectId, countryId, deleteAccountRequested }: UserPreferences) => {
+      if (!projectId && !countryId && !deleteAccountRequested) {
```

Comment:
```suggestion
      if (!projectId && !countryId && deleteAccountRequested === undefined) {
```
Since it's a boolean, it could theoretically be passed in as false and the update wouldn't run as expected. There isn't a use case in UI for setting to false but it's possible there could be in the future and this could lead to a bug.